src/Core/Containers/VectorArray.hpp

```diff
@@ -3,21 +3,34 @@
 #include <Core/Containers/AlignedStdVector.hpp>
 #include <Core/RaCore.hpp>
 
+#include <type_traits> // std::is_integral
+
 namespace Ra {
 namespace Core {
 
+/// Internal class used to compute types and values used by VectorArray, to handle both
+/// eigen matrices and integral types
+template <typename V, bool isIntegral>
+struct VectorArrayTypeHelper {};
+
 /// This class is a wrapper around a std::vector of Core::Vectors.
 /// which allow to use the stdlib's dynamic array implementation, yet pass it as
 /// a matrix when Eigen needs it with the getMap() method.
 template <typename V>
 class VectorArray : public AlignedStdVector<V>
 {
+  private:
+    using TypeHelper =
+        VectorArrayTypeHelper<V, std::is_integral<V>::value || std::is_floating_point<V>::value>;
+
   public:
     // Type shortcuts
-    using Vector         = V;
-    using Matrix         = Eigen::Matrix<typename V::Scalar, V::RowsAtCompileTime, Eigen::Dynamic>;
-    using MatrixMap      = Eigen::Map<Matrix>;
-    using ConstMatrixMap = Eigen::Map<const Matrix>;
+    static constexpr int ElementSize = TypeHelper::Size;
+    using Scalar                     = typename TypeHelper::Scalar;
```

Comment:
I prefer something like `value_type` instead of `Scalar` to prevent confusion with the Radium `Scalar` type definition (i.e. `float` or `double` depending on the building options)
```suggestion
    using value_type                     = typename TypeHelper::value_type;
```